docs/python-extensive/data/api.md

```diff
@@ -0,0 +1,469 @@
+solved the quiz question.
+
+??? info "URL construction"
+
+    ```python
+    api_url = "https://api.coincap.io/v2"
+    coin_id = "pepe-cash"
+    endpoint = f"/assets/{coin_id}/history"
+    query_params = "?interval=d1"  # daily interval
+    
+    url = f"{api_url}{endpoint}{query_params}"
+    ```
+    
+    Let's walk through the URL construction step by step:
+    
+    1. `api_url` is the base URL of the API (nothing new here).
+    2. `coin_id` :fontawesome-solid-arrow-right: `pepe-cash` is the 
+        identifier of the cryptocurrency we want to retrieve data 
+        for. We have already requested all cryptocurrency identifiers like 
+        `bitcoin` or `ethereum` with the `/assets` endpoint. Have another look 
+        at the table [here](#endpoints-continued).
+    3. `endpoint` contains the endpoint name we want to access, in this 
+        particular case :fontawesome-solid-arrow-right: 
+        `/assets/pepe-cash/history`.
+    4. `query_params` stands for ==query parameters== which are additional 
+        parameters that are passed to the server. Think of a `Python` 
+        function with the endpoint being the function name and the query 
+        parameters being the function parameters used for fine-grained control.
+
+        Query parameters are separated from the URL by a `?`.
+        In this case, we specified `?interval=d1`. `interval` is the 
+        parameter name followed by the value `d1` which stands for daily 
+        price history intervals. Again, with a `Python` function you can think 
+        of `interval=d1` as a named argument.
+    
+        More detailed explanations on both parameters and values are 
+        specified in the API documentation.
+    
+    Finally, we end up with the URL 
+    `#!python "https://api.coincap.io/v2/assets/pepe-cash/history?interval=d1"`
+    
+    <div style="text-align: center;">
+        <iframe src="https://giphy.com/embed/l1Etfpt5pdYl34BuU" width="480" height="360" style="" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/spongebob-spongebob-squarepants-season-5-l1Etfpt5pdYl34BuU"></a></p>
+        <figcaption>Quite a complicated URL.</figcaption>
+    </div>
+
+### Request
+
+If you've followed the construction of the URL closely, we can easily send 
+another request to retrieve market data. This time around it is another `GET`
+request, however with a query parameter.
+
+```python
+# construct the URL (same as above)
+api_url = "https://api.coincap.io/v2"
+coin_id = "pepe-cash"
+endpoint = f"/assets/{coin_id}/history"
+query_params = "?interval=d1"  # daily interval
+
+url = f"{api_url}{endpoint}{query_params}"
+
+# send the request
+response = requests.get(url=url)
+```
+
+Again, convert the response to a `DataFrame` and print the first few rows.
+
+```python
+pepe_history = response.json()
+pepe_history = pd.DataFrame(pepe_history["data"])
+
+print(pepe_history.head())
```

Comment:
head() visualizes the oldest data points. maybe it is a better idea to use tail() to retrieve the current prices?

Reply:
Yep, you're right 👍